docs/registry/README.md

```diff
@@ -0,0 +1,385 @@
+there is no way of doing exact matching of Cloud Event sources on Triggers. Thus, we propose adding this custom extension 
+to Cloud Events whenever we can. If the extension is not present, then we fallback to the Cloud Event source. 
+Note that when we start supporting more advanced filtering mechanisms on Triggers, we might not need this.
+
+
+- `schema` is a URI with the EventType schema. It may be a JSON schema, a protobuf schema, etc. It is optional.
+
+- `broker` refers to the Broker that can provide the EventType. 
+
+### Typical Flow
+
+`1.` A `Cluster Configurator` configures the cluster in a way that allows the population of EventTypes in the Registry. 
+We foresee the following three ways of populating the Registry so far:
+
+`1.1` Event Source CR installation
+
+Upon installation of an Event Source CR by a `Cluster Configurator`, the Source will register its EventTypes.
+
+Example:
+
+```yaml
+apiVersion: sources.eventing.knative.dev/v1alpha1
+kind: GitHubSource
+metadata:
+  name: github-source-sample
+  namespace: default
+spec:
+  eventTypes:
+    - push
+    - pull_request
+  ownerAndRepository: my-other-user/my-other-repo
+  accessToken:
+    secretKeyRef:
+      name: github-secret
+      key: accessToken
+  secretToken:
+    secretKeyRef:
+      name: github-secret
+      key: secretToken
+  sink:
+    apiVersion: eventing.knative.dev/v1alpha1
+    kind: Broker
+    name: default
+
+```
+ 
+By applying the above file, two EventTypes will be registered, with types `dev.knative.source.github.push` and 
+`dev.knative.source.github.pull_request`, source `my-other-user/my-other-repo`, for the `default` Broker in the `default`
+ namespace, and with owner `github-source-sample`.
+ 
+Note that the `Cluster Configurator` is the person in charge of taking care of authentication-related matters. E.g., if a new `Event Consumer` 
+wants to listen for events from a different GitHub repo, the `Cluster Configurator` will take care of the necessary secrets generation, 
+and new Source instantiation.
+ 
+In YAML, the above EventTypes would look something like these:
+
+```yaml
+apiVersion: eventing.knative.dev/v1alpha1
+kind: EventType
+metadata:
+  generateName: dev.knative.source.github.push-
+  namespace: default
+  owner: # Owned by github-source-sample
+spec:
+  type: dev.knative.source.github.push
+  source: my-other-user/my-other-repo
+  broker: default
+---
+apiVersion: eventing.knative.dev/v1alpha1
+kind: EventType
+metadata:
+  generateName: dev.knative.source.github.pullrequest-
+  namespace: default
+  owner: # Owned by github-source-sample
+spec:
+  type: dev.knative.source.github.pull_request
+  source: my-other-user/my-other-repo
+  broker: default
+```
+
+Two things to notice: 
+- We generate the names by stripping invalid characters from the original type (e.g., `_`)
+- The `spec.type` adds the prefix `dev.knative.source.github.` This is a **separate discussion** on whether we should 
```

Comment:
where did `dev.knative.source.github` come from? In particuarl, if `dev.knative.source` is static, I'm not clear on where `github` came from since that string (by itself) is nowhere in the CRD.

Reply:
That is added in the GitHubSource controller, as the adapter adds that prefix. 
As of now, I haven't changed anything on the CRD, so it uses github naming (e.g., pull_request). 
You would rather add the dev.knative.github.source.github prefix to all types in the CRD? And then strip that off when creating the webhook?